docs/src/embeddings/custom_embedding_function.md

```diff
@@ -15,198 +15,225 @@ There is another optional layer of abstraction available: `TextEmbeddingFunction
-
-    def ndims(self):
-        if self._ndims is None:
-            self._ndims = len(self.generate_embeddings("foo")[0])
-        return self._ndims
-
-    @cached(cache={}) 
-    def _embedding_model(self):
-        return sentence_transformers.SentenceTransformer(name)
-```
-
-This is a stripped down version of our implementation of `SentenceTransformerEmbeddings` that removes certain optimizations and defaul settings.
+
+=== "Python"
+
+    ```python
+    from lancedb.embeddings import register
+    from lancedb.util import attempt_import_or_raise
+
+    @register("sentence-transformers")
+    class SentenceTransformerEmbeddings(TextEmbeddingFunction):
+        name: str = "all-MiniLM-L6-v2"
+        # set more default instance vars like device, etc.
+
+        def __init__(self, **kwargs):
+            super().__init__(**kwargs)
+            self._ndims = None
+
+        def generate_embeddings(self, texts):
+            return self._embedding_model().encode(list(texts), ...).tolist()
+
+        def ndims(self):
+            if self._ndims is None:
+                self._ndims = len(self.generate_embeddings("foo")[0])
+            return self._ndims
+
+        @cached(cache={})
+        def _embedding_model(self):
+            return sentence_transformers.SentenceTransformer(name)
+    ```
+
+=== "TypeScript"
+
+    ```ts
+    --8<--- "nodejs/examples/custom_embedding_function.ts:imports"
+    --8<--- "nodejs/examples/custom_embedding_function.ts:embedding_impl"
+    ```
+
+
+This is a stripped down version of our implementation of `SentenceTransformerEmbeddings` that removes certain optimizations and default settings.
 
 Now you can use this embedding function to create your table schema and that's it! you can then ingest data and run queries without manually vectorizing the inputs.
 
-```python
-from lancedb.pydantic import LanceModel, Vector
+=== "Python"
+
+    ```python
+    from lancedb.pydantic import LanceModel, Vector
+
+    registry = EmbeddingFunctionRegistry.get_instance()
+    stransformer = registry.get("sentence-transformers").create()
 
-registry = EmbeddingFunctionRegistry.get_instance()
-stransformer = registry.get("sentence-transformers").create()
+    class TextModelSchema(LanceModel):
+        vector: Vector(stransformer.ndims) = stransformer.VectorField()
+        text: str = stransformer.SourceField()
 
-class TextModelSchema(LanceModel):
-    vector: Vector(stransformer.ndims) = stransformer.VectorField()
-    text: str = stransformer.SourceField()
+    tbl = db.create_table("table", schema=TextModelSchema)
 
-tbl = db.create_table("table", schema=TextModelSchema)
+    tbl.add(pd.DataFrame({"text": ["halo", "world"]}))
+    result = tbl.search("world").limit(5)
+    ```
 
-tbl.add(pd.DataFrame({"text": ["halo", "world"]}))
-result = tbl.search("world").limit(5)
-```
+=== "TypeScript"
+
+    ```ts
+    --8<--- "nodejs/examples/custom_embedding_function.ts:call_custom_function"
+    ```
 
 NOTE:
```

Comment:
Not related to this PR, but would you mind fixing this so it uses a proper `!!! note` admonition?